File: src/test/java/org/goafabric/calleeservice/logic/CalleeLogicTest.java
package org.goafabric.calleeservice.logic;

import org.goafabric.calleeservice.controller.dto.Callee;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class CalleeLogicTest {
    private CalleeLogic calleeLogic = new CalleeLogic();

    @Test
    void sayMyName() {
        assertThat(calleeLogic.sayMyName("Heisenberg").message())
                .isEqualTo("Your name is: Heisenberg");
    }

    @Test
    void sayMyOtherName() {
        assertThat(calleeLogic.sayMyOtherName("Slim Shady").message())
                .isEqualTo("Your other name is: Slim Shady");
    }

    @Test
    void save() {
        assertThat(calleeLogic.save(new Callee("0", "saved by the bell")).message())
                .isEqualTo("Storing your message: saved by the bell");
    }
}

File: src/test/kotlin/org/goafabric/calleeservice/logic/CalleeLogicTest.kt
package org.goafabric.calleeservice.logic

import org.assertj.core.api.Assertions.assertThat
import org.goafabric.calleeservice.controller.dto.Callee
import org.junit.jupiter.api.Test

internal class CalleeLogicTest {
    private val calleeLogic = CalleeLogic()

    @Test
    fun sayMyName() {
        assertThat(calleeLogic.sayMyName("Heisenberg").message)
            .isEqualTo("Your name is: Heisenberg")
    }

    @Test
    fun sayMyOtherName() {
        assertThat(calleeLogic.sayMyOtherName("Slim Shady").message)
            .isEqualTo("Your other name is: Slim Shady")
    }

    @Test
    fun save() {
        assertThat(calleeLogic.save(Callee("0", "saved by the bell"))!!.message)
            .isEqualTo("Storing your message: Callee(id=0, message=saved by the bell).message")
    }
}